src/interfaces/multi_particle_distribution.jl

```diff
@@ -0,0 +1,137 @@
+
+"""
+
+   MultiParticleDistribution
+
+Base type for sample drawing from multiple particle distributions. The following interface functions
+should be implemented:
+
+* [`QEDevents._particles(d::MultiParticleDistribution)`](@ref)
+* [`QEDevents._particle_directions(d::MultiParticleDistribution)`](@ref)
+* [`QEDevents.randmom(rng::AbstractRNG,d::MultiParticleDistribution)`](@ref)
+
+"""
+const MultiParticleDistribution = ParticleSampleable{MultiParticleVariate}
+
+Broadcast.broadcastable(d::MultiParticleDistribution) = Ref(d)
+
+Base.length(d::MultiParticleDistribution) = length(_particles(d))
+Base.size(d::MultiParticleDistribution) = (length(d),)
+
+"""
+    _particle(dist::MultiParticleDistribution)
+
+Return tuple of particles associated with the `dist`.
+
+!!! note
+
+    Interface function to be implemented for multi-particle distributions.
+
+"""
+function _particles end
+
+"""
+    _particle_direction(dist::MultiParticleDistribution)
+
+Return tuple of particle-directions for all particles associated with `dist`.
+
+!!! note
+
+    Interface function to be implemented for multi-particle distributions.
+
+"""
+function _particle_directions end
+#default
+function _particle_direction(d::MultiParticleDistribution)
+    return Tuple(fill(UnknownDirection(), length(d)))
+end
+
+"""
+
+    randmom(rng::AbstractRNG,d::MultiParticleDistribution)
+
+Return an iteratable container (e.g. vector or tuple) of momenta according to the distribution `d`.
+"""
+function randmom end
```

Comment:
```suggestion
"""

    rand_mom(rng::AbstractRNG, d::MultiParticleDistribution)

Return an iterable container (e.g. vector or tuple) of momenta according to the distribution `d`.
"""
function rand_mom end
```

I know functions such as `is<thing>` are supposed to be without an underscore according to the Julia style guide, but does this apply here? I feel like `rand_mom(...)` would look nicer.

Reply:
I was following the convention of the Julia standard library: there they have `randn` and `randexp` without the underscore.

Reply:
Okay

Reply:
Approved by authority 😅

Reply:
I mean if it's part of the style guide then that's how it should be. I was just unaware of that :+1: